### PR TITLE
Support primitive types for SQS DataType

### LINKED ISSUE #221: SQS Message Attributes - Number types
When a message attribute includes a Number type but does not specify the **_optional_** custom type, the method [getNumberValue(MessageAttributesValue) in QueueMessageUtils (line 93)](https://github.com/spring-cloud/spring-cloud-aws/blob/master/spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/core/QueueMessageUtils.java) causes a StringIndexOutOfBoundsException when trying to parse the "numberTypeClass" out of the DataType field of the attribute. 

Since this is an [optional part of the DataType field in the MessageAttribute](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-attributes.html#message-attributes-data-types-validation), the code should gracefully handle when it is not present instead of blindly trying to parse it out. If it is not present, you could default to parsing the value as a long or whatever numeric type seems to be appropriate as a default.

This is a blocker for me at the moment, since the [library that is putting the message in the queue ](https://github.com/awslabs/amazon-sqs-java-extended-client-lib) is not in my control and is not specifying the optional piece as part of the DataType of the message attribute. You could potentially argue he should be adding that number class type, but I would think that Spring AWS Cloud should be more resilient than just failing to handle the message.

### PR DESCRIPTION
**Pull request by joshuaali**
As well as supporting primitive types, this also makes the optional custom type just that: optional.

Resolves #221.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@joshuaali Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-aws&pullRequestId=293)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-aws&pullRequestId=293) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/support/converter/NotificationRequestConverter.java [hunk 1]
```diff
@@ -120,12 +123,9 @@ public Object getMessage() {
...
                 } else if (attributeType.startsWith(MessageAttributeDataTypes.NUMBER)) {
-                    Object numberValue = getNumberValue(attributeType, attributeValue);
-                    if (numberValue != null) {
-                        messageHeaders.put(attributeName, numberValue);
-                    }
-                } else if (MessageAttributeDataTypes.BINARY.equals(attributeName)) {
-                    messageHeaders.put(attributeName, ByteBuffer.wrap(attributeType.getBytes()));
+                    messageHeaders.put(attributeName, getNumberValue(attributeType, attributeValue));
+                } else if (MessageAttributeDataTypes.BINARY.equals(attributeType)) {
+                    messageHeaders.put(attributeName, ByteBuffer.wrap(attributeValue.getBytes()));
```
Comment (joshuaali) [inline spring-cloud-aws-messaging/src/main/java/org/springframework/cloud/aws/messaging/support/converter/NotificationRequestConverter.java]: Whilst going through this code, I noticed something peculiar: `MessageAttributeDataTypes.BINARY` was being compared with `attributeName` and the value was taken from `attributeType`.

**Comment by pivotal-issuemaster** [comment]
@joshuaali Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-aws&pullRequestId=293)!

**Comment by the-fine** [comment]
Any chance this PR will be merged any time soon?

**Comment by berrueta** [comment]
I've hit this bug too. How can I help to have this PR merged?

**Comment by scottscott** [comment]
It looks like this has passed the one year mark and has no reviewers. Can this be reviewed and hopefully merged?

**Comment by t0il3ts0ap** [comment]
Could this be reviewed and merged?

**Comment by maciejwalkowiak** [comment]
@joshuaali thank you for contribution and apologies your PR has not been tackled on time. The issue is fixed now.

**Pull request closed without merging**